improve timeout handling

mattp- opened the pull request:
the timeout kwarg previously was only being used for polling timeout. we
can also pass it as a low kwarg and use it for urllib connection timeout
as well.

mattp- committed:
improve timeout handling

the timeout kwarg previously was only being used for polling timeout. we
can also pass it as a low kwarg and use it for urllib connection timeout
as well.

mattp- committed:
emit pepper-runtets.log for debug

mattp- committed:
revert previous change, add a buffer on local timeout

since these timeout kwargs should coincide with upstream response, we
give a bit of buffer to let salt-api time out gracefully before forcing
it first.

mattp- committed:
temporarily disable to see what travis does

eplodn commented:
Will this solve #198 ?

mattp- commented:
@eplodn no, I think what you're actually looking for is https://github.com/saltstack/pepper/blob/develop/pepper/cli.py#L647
its gated off because its not compatible with tornado salt-api, but if you run cherrypy it will work as is. the pr to fix it has been sitting for a few months in salts queue

eplodn commented:
I'm using pepper 0.7.5 and salt 2018.3.3 with cherrypy and it doesn't work, what am I missing?

mattp- commented:
@eplodn timeouts in salt are tricky, there is no 'timeout' in the sense you are thinking for a long running job that will abort. if a minion is responding to the master polling for status, it will run forever. This PR should hypothetically do something on the client side for what you are looking for but it it is not ready for merge atm

mattp- committed:
improve guessing game for runner outputter

mattp- committed:
make logging verbosity actually work

mattp- committed:
forced full_return: move to parse_cmd to restore --json behavior

--json was broken with the introduction of this because client isnt set
/ shortcircuited when json is used. We can add the equivalent logic in parse_cmd
instead.

mattp- committed:
runners: use arg/kwarg instead of popping things into low data

mixing low data with kwargs has been deprecated forever, it should be
safe to change this now.

nicholasmhughes commented:
Howdy! :wave: 

Apologies for the long delay on this integrating this code. We certainly appreciate your effort!

Pepper is getting some renewed focus and love, and we're checking in to see if you'd like to update this against the latest version of the base branch and ensure that all merge conditions are met.

If the items addressed by this contribution have already been fixed/added, are no longer an issue, or you no longer have a desire to work on this... please let us know and/or close out this PR.

If we don't hear from you within 7 days, we'll assume the code has been abandoned and close it out. If you'd like to revisit this in the future, you can always open a new PR.

Thank you, and we hope to see you active in the Salt ecosystem in the future!
